### PR TITLE
Handle cases where not all actions can be delegated

### PR DESCRIPTION
**Pull request by sonwit**
Handle cases where not all actions can be delegated

## Description
Add info message to delegation-modal 
Add test-cases to storybook 

## Related Issue(s)
- #1085 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)
Skjermbilde 2024-10-23 kl  16 18 00
Skjermbilde 2024-10-23 kl  16 17 50

### CODE REVIEW

File: src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx [hunk 1]
```diff
@@ -49,22 +49,50 @@ export const ResourceInfo = ({ resource, toParty, onDelegate }: ResourceInfoProp
...
   const resourceRef: ResourceReference | null =
     resource !== undefined
       ? {
           resource: resource.authorizationReference,
         }
       : null;
 
+  const getMissingAccessMessage = (response: DelegationAccessResult[]) => {
+    const hasMissingRoleAccess = response.some((result) =>
+      result.details?.some((detail) => detail.code === 'MissingRoleAccess'),
```
Comment (allinox) [inline src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx:69]:
```suggestion
      result.details?.some((detail) => detail.code === 'MissingRoleAccess' || detail.code === 'MissingRightAccess'),
```
There are two cases that should result in the same message to the user

File: src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx [hunk 2]
```diff
@@ -49,22 +49,50 @@ export const ResourceInfo = ({ resource, toParty, onDelegate }: ResourceInfoProp
...
+    );
+
+    const hasMissingSrrRightAccess = response.some(
+      (result) =>
+        !hasMissingRoleAccess &&
+        result.details?.some((detail) => detail.code === 'MissingSrrRightAccess'),
```
Comment (allinox) [inline src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx:75]: Also, consider using the `ErrorCode` enum so we are less vulnerable to things like typos in string values

**Review by allinox — APPROVED** [review]
Another case covered - great! 😄

I left a couple of suggestions, but as usual, I'm gonna approve it so you can just merge it when you've gone through them 😊

File: src/localizations/no_nb.json [hunk 3]
```diff
@@ -228,7 +228,9 @@
       "time_of_error": "Tidspunkt feilen inntraff: {{time}}",
       "error_status": "Status: {{status}}",
       "access_list_service": "Hvem som får benytte denne tjenesten bestemmes av {{resourceOwner}}, og {{reportee}} er ikke gitt den muligheten. Vi anbefaler dere å ta kontakt med {{resourceOwner}}.",
-      "missing_rights": "Du får ikke gitt denne fullmakten til andre fordi du mangler den selv. Daglig leder eller hovedadministrator kan hjelpe deg med dette."
+      "missing_rights": "Du får ikke gitt denne fullmakten til andre fordi du mangler den selv. Daglig leder eller hovedadministrator kan hjelpe deg med dette.",
+      "missing_role_access_message": "Du får ikke delegert hele tjenesten, fordi du mangler noen av rettighetene selv. Daglig leder eller hovedadministrator kan gi deg tilgang.",
+      "missing_srr_right_access_message": "{{reportee}} er ikke gitt tillatelse fra {{resourceOwner}} til å delegere hele tjenesten. Ta kontakt med dem dersom du ønsker å endre dette. "
```
[the remark this answers is not in the document]
Comment (sonwit) [reply src/localizations/no_nb.json]: 👍

File: src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx (hunk 2 above)
Comment (sonwit) [reply src/features/amUI/userRightsPage/DelegationModal/ResourceInfo.tsx]: 👍

**Pull request merged**